sdk/core/core-amqp/review/core-amqp.api.md

```diff
@@ -22,6 +22,12 @@ import { SenderOptions } from 'rhea-promise';
 import { Session } from 'rhea-promise';
 import { WebSocketImpl } from 'rhea-promise';
 
+// @public
+export interface AcquireOptions {
+    abortSignal?: AbortSignalLike;
```

Comment:
Let's just make the abortSignal non-optional so you have to explicitly opt out of it. I think, in our cases, it's pretty much always going to be a mistake.

Reply:
@richardpark-msft, 
The problem with that is an interface named "*Options" is expected to have only optional stuff. .

@chradek, 
Regarding the name, consider AcquireLockOptions instead. Since this gets exported from core-amqp, the word "lock" will provide the context needed.

Reply:
With regards to optional vs non-optional. I do agree it's beneficial to have optional things marked as required just to ensure that we really are passing a value.

Where I drew the line though was with the public API, which I believe @richardpark-msft agrees with. Since this is in a core package that we don't expect customers to use directly, there's an argument that we can treat this more like an internal API. If I changed the interface to `AcquireLockProperties` and made it required, would that make everyone happy?

Reply:
I can live with AcquireLockProperties having non optional stuff

Reply:
Renamed to AcquireLockProperties and made fields mandatory.